src/platforms/esp32/main/ledc_nifs.c

```diff
@@ -0,0 +1,503 @@
+    VALIDATE_VALUE(timer_sel, term_is_integer);
+    term duty = interop_proplist_get_value(config, context_make_atom(ctx, ledc_duty));
+    VALIDATE_VALUE(duty, term_is_integer);
+    term hpoint = interop_proplist_get_value(config, context_make_atom(ctx, ledc_hpoint));
+    VALIDATE_VALUE(hpoint, term_is_integer);
+
+    ledc_channel_config_t ledc_channel = {
+        .gpio_num = term_to_int(gpio_num),
+        .speed_mode = term_to_int(speed_mode),
+        .channel = term_to_int(channel),
+        .timer_sel = term_to_int(timer_sel),
+        .duty = term_to_int(duty),
+        .hpoint = term_to_int(hpoint)
+    };
+    esp_err_t err = ledc_channel_config(&ledc_channel);
+    switch (err) {
+        case ESP_OK:
+            TRACE("ledc_channel_config\n");
+            return OK_ATOM;
+        default:
+            fprintf(stderr, "Unable to set ledc channel config. err=%i\n", err);
+            RAISE_ERROR(term_from_int(err));
+    }
+}
+
+static term nif_ledc_fade_func_install(Context *ctx, int argc, term argv[])
+{
+    UNUSED(argc);
+    term flags = argv[0];
+    VALIDATE_VALUE(flags, term_is_integer);
+
+    esp_err_t err = ledc_fade_func_install(term_to_int(flags));
+    switch (err) {
+        case ESP_OK:
+            TRACE("ledc_fade_func_install\n");
+            return OK_ATOM;
+        default:
+            fprintf(stderr, "Unable to install fade func. err=%i\n", err);
+            RAISE_ERROR(term_from_int(err));
+    }
+}
+
+static term nif_ledc_fade_func_uninstall(Context *ctx, int argc, term argv[])
+{
+    UNUSED(argc);
+    UNUSED(argv);
+
+    ledc_fade_func_uninstall();
+    TRACE("ledc_fade_func_install\n");
+    return OK_ATOM;
+}
+
+static term nif_ledc_set_fade_with_time(Context *ctx, int argc, term argv[])
+{
+    UNUSED(argc);
+    term speed_mode = argv[0];
+    VALIDATE_VALUE(speed_mode, term_is_integer);
+    term channel = argv[1];
+    VALIDATE_VALUE(channel, term_is_integer);
+    term duty = argv[2];
+    VALIDATE_VALUE(duty, term_is_integer);
+    term fade_time = argv[3];
+    VALIDATE_VALUE(fade_time, term_is_integer);
+
+    esp_err_t err = ledc_set_fade_with_time(
+        term_to_int(speed_mode),
+        term_to_int(channel),
+        term_to_int(duty),
+        term_to_int(fade_time)
+    );
+    switch (err) {
+        case ESP_OK:
+            TRACE("ledc_set_fade_with_time\n");
+            return OK_ATOM;
+        default:
+            fprintf(stderr, "Unable to ledc_set_fade_with_time. err=%i\n", err);
+            RAISE_ERROR(term_from_int(err));
+    }
+}
+
+static term nif_ledc_set_fade_with_step(Context *ctx, int argc, term argv[])
+{
+    UNUSED(argc);
+    term speed_mode = argv[0];
+    VALIDATE_VALUE(speed_mode, term_is_integer);
+    term channel = argv[1];
+    VALIDATE_VALUE(channel, term_is_integer);
+    term duty = argv[2];
+    VALIDATE_VALUE(duty, term_is_integer);
+    term scale = argv[3];
+    VALIDATE_VALUE(scale, term_is_integer);
+    term cycle_num = argv[3];
```

Comment:
oops

Reply:
oops what? :D

Reply:
term cycle_num = argv[3]; should be term cycle_num = argv[4];